### PR TITLE
Fixes for deleted lots and ordering

### LINKED ISSUE #1793: Eigen needs to handle lot deletion elegantly
Issue: 
Lots get deleted from auctions pretty often. We observed during Phillips that it happendd last minute and a lot. Eigen did not handle this well when we passed a deleted lot our "current lot" logic to show the user which lot we are at was incorrect. When the user hit the "Jump to current lot" marker they were taken to lot n+1. This gets fixed over time as the auction progressed but we need to guard against this as we anticipate this happening a lot.

### PR DESCRIPTION
**Pull request by ashfurrow**
Two things:

- [x] I couldn't reproduce the problem, so it could have been something fixed on the server in the mean time, or I just haven't reproduced the actual bug yet.
- [x] If the model's position is not reliable as its index in the lots list, then I should look for other problems that might cause.
- [x] Need to test changes still.

Fixes #1793.

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
  </tbody>
</table>

Time | Class | Function |
| --- | ----- | ----- |238.2ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-deleted-lot/Artsy/View_Controllers/Auction/AuctionViewController.swift#L199) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
204.7ms | [SwiftyJSON.swift](/artsy/eigen/tree/ashfurrow-deleted-lot/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
172.3ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-deleted-lot/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
88.1ms | [Observable.swift](/artsy/eigen/tree/ashfurrow-deleted-lot/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
71.2ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/ashfurrow-deleted-lot/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, salesPerson: LiveAuctionsSalesPersonType, sideMargin: String)
71.1ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/ashfurrow-deleted-lot/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
63.2ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/ashfurrow-deleted-lot/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L25) | func ar_presentBlurredOverlayWithTitle(title: String, subtitle: String, buttonState: BlurredStatusOverlayViewCloseButtonState = default)
63.0ms | [LiveAuctionLotViewController.swift](/artsy/eigen/tree/ashfurrow-deleted-lot/Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift#L49) | @objc override func viewDidLoad()
61.2ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/ashfurrow-deleted-lot/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L31) | (closure)
56.8ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/ashfurrow-deleted-lot/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionLotSetViewController.swift [hunk 1]
```diff
@@ -69,13 +69,17 @@ class LiveAuctionLotSetViewController: UIViewController {
     var suppressJumpingToOpenLots = false
 
     func hasChangedLot(lot: LiveAuctionLotViewModelType?) {
-        guard let newLot = lot where !suppressJumpingToOpenLots else { return }
+        guard let
+            newLot = lot,
+            newLotIndex = salesPerson.indexForViewModel(newLot)
+            where !suppressJumpingToOpenLots
+            else { return }
```
Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionLotSetViewController.swift]: This is the part I'm most keen to test.

File: CHANGELOG.yml [hunk 2]
```diff
@@ -11,6 +11,7 @@ upcoming:
       - Fallback to higher or lower available artwork image. - alloy
       - Fixes problem where users looking at a closed live auction would be directed to the live interface. - ash
       - Fixes problem where users weren't directed to live auction when it started. - ash
+      - Fixes problems going to the current lot of a live auction. - ash
```
Comment (orta) [inline CHANGELOG.yml]: Can you expand on this? (it doesn't really match the title )

Comment (ashfurrow) [inline CHANGELOG.yml]: Fair enough, we were relying on a sale artwork's `position` property to be the same as its position in the `saleArtworks` array, which it isn't guaranteed to be. I'll expand the entry.

**Comment by orta** [comment]
👍 good to go on green

**Pull request merged**